tests/test_processing_rt.py

```diff
@@ -0,0 +1,258 @@
+        axes_name="xy",
+        axis_to_recalculate="y",
+    )
+
+    rt_xy_expected = Rototrans(
+        np.array(
+            [
+                [0.5773502691896257, 0.7071067811865475, -0.408248290463863, 1.0],
+                [0.5773502691896257, 0.0, 0.816496580927726, 2.0],
+                [0.5773502691896257, -0.7071067811865475, -0.408248290463863, 3.0],
+                [0, 0, 0, 1.0],
+            ]
+        )
+    )
+
+    np.testing.assert_array_equal(rt_xy_from_known_m, rt_xy_expected)
+
+    exception_default_params = dict(
+        origin=all_m.isel(channel=[0]),
+        axis_1=all_m.isel(channel=[0, 1]),
+        axis_2=all_m.isel(channel=[0, 2]),
+        axes_name="xy",
+        axis_to_recalculate="y",
+    )
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(origin=all_m.isel(channel=[0, 1]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_1=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_2=all_m.isel(channel=[0]))}
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{
+                **exception_default_params,
+                **dict(axis_1=all_m.isel(channel=[0, 1], time=slice(None, 50))),
+            }
+        )
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="yyz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="xxz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(**{**exception_default_params, **dict(axes_name="zzz")})
+
+    with pytest.raises(ValueError):
+        Rototrans.from_markers(
+            **{**exception_default_params, **dict(axis_to_recalculate="h")}
+        )
+
+
+def test_rt_transpose():
+    n_frames = 10
+    angles = Angles.from_random_data(size=(3, 1, n_frames))
+    rt = Rototrans.from_euler_angles(angles, angle_sequence="xyz")
+
+    rt_t = Rototrans.from_transposed_rototrans(rt)
+
+    rt_t_expected = np.zeros((4, 4, n_frames))
+    rt_t_expected[3, 3, :] = 1
+    for row in range(rt.row.size):
+        for col in range(rt.col.size):
+            for frame in range(rt.time.size):
+                rt_t_expected[col, row, frame] = rt[row, col, frame]
+
+    for frame in range(rt.time.size):
+        rt_t_expected[:3, 3, frame] = -rt_t_expected[:3, :3, frame].dot(
+            rt[:3, 3, frame]
+        )
+
+    np.testing.assert_array_almost_equal(rt_t, rt_t_expected, decimal=10)
+
+
+def test_average_rt():
+    # TODO: investigate why this does not work
+    # angles = Angles.from_random_data(size=(3, 1, 100))
+    # or
+    # angles = Angles(np.arange(300).reshape((3, 1, 100)))
```

Comment:
Is this solved?

Reply:
Yes

Reply:
Done in https://github.com/pyomeca/pyomeca/pull/88/commits/01e49cb7f71fb877049917d3c17d80396cdd17c1